packages/context/src/inject-config.ts

```diff
@@ -0,0 +1,167 @@
+   * expect(store2.optionX).to.eql(2);
+   * expect(store2.optionY).to.eql('b');
+   * ```
+   *
+   * @param configPath Optional property path of the config. If is `''` or not
+   * present, the `config` object will be returned.
+   * @param metadata Optional metadata to help the injection
+   */
+export function config(configPath?: string, metadata?: InjectionMetadata) {
+  configPath = configPath || '';
+  metadata = Object.assign(
+    {configPath, decorator: '@config', optional: true},
+    metadata,
+  );
+  return inject('', metadata, resolveFromConfig);
+}
+
+export namespace config {
+  export const getter = function injectConfigGetter(
+    configPath?: string,
+    metadata?: InjectionMetadata,
+  ) {
+    configPath = configPath || '';
+    metadata = Object.assign(
+      {configPath, decorator: '@config.getter', optional: true},
+      metadata,
+    );
+    return inject('', metadata, resolveAsGetterFromConfig);
+  };
+
+  export const view = function injectConfigView(
+    configPath?: string,
+    metadata?: InjectionMetadata,
+  ) {
+    configPath = configPath || '';
+    metadata = Object.assign(
+      {configPath, decorator: '@config.view', optional: true},
+      metadata,
+    );
+    return inject('', metadata, resolveAsViewFromConfig);
+  };
+}
+
+function resolveFromConfig(
+  ctx: Context,
+  injection: Injection,
+  session: ResolutionSession,
+): ValueOrPromise<unknown> {
+  if (!session.currentBinding) {
+    // No binding is available
+    return undefined;
+  }
+
+  const meta = injection.metadata || {};
+  const binding = session.currentBinding;
+
+  return ctx.getConfigAsValueOrPromise(binding.key, meta.configPath, {
+    session,
+    optional: meta.optional,
+  });
+}
+
+function resolveAsGetterFromConfig(
+  ctx: Context,
+  injection: Injection,
+  session: ResolutionSession,
+) {
+  if (!session.currentBinding) {
+    // No binding is available
+    return undefined;
+  }
+  const meta = injection.metadata || {};
+  const bindingKey = session.currentBinding.key;
+  // We need to clone the session for the getter as it will be resolved later
+  const forkedSession = ResolutionSession.fork(session);
+  return async function getter() {
+    return ctx.getConfigAsValueOrPromise(bindingKey, meta.configPath, {
+      session: forkedSession,
+      optional: meta.optional,
+    });
+  };
+}
+
+function resolveAsViewFromConfig(
+  ctx: Context,
+  injection: Injection,
+  session: ResolutionSession,
+) {
+  const targetType = inspectTargetType(injection);
+  if (targetType && targetType !== ContextView) {
+    const targetName = ResolutionSession.describeInjection(injection)!
+      .targetName;
+    throw new Error(
+      `The type of ${targetName} (${targetType.name}) is not ContextView`,
+    );
+  }
+  if (!session.currentBinding) {
+    // No binding is available
+    return undefined;
+  }
+  const bindingKey = session.currentBinding.key;
+  const view = new ConfigView(
+    ctx,
+    binding =>
```

Comment:
Can you cache this filter function in `Injection` please? So that we create the filter function only once when `@config.view()` is invoked, instead of re-creating it every time a dependency is resolved/injected.

Reply:
Please note the filter function depends on the current binding key.